### PR TITLE
Fix problem with special permission

### LINKED ISSUE #406: PD does not generate kt version of SYSTEM_ALERT_WINDOW
PD version is 3.0.1

if you set `NeedsPermission(SYSTEM_ALERT_WINDOW)`, it won't generate code.

This is because `SystemAlertWindowHelper` use JavaPoet's ClassName.

https://github.com/permissions-dispatcher/PermissionsDispatcher/blob/master/processor/src/main/kotlin/permissions/dispatcher/processor/impl/kotlin/SystemAlertWindowHelper.kt#L3

Also, I guess `onActivityResult` method should be extension method of Activity.

I will fix these and add test cases.

### PR DESCRIPTION
**Pull request by shiraji**
Close #406

### CODE REVIEW

File: test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    private lateinit var activity: ActivityWithSystemAlertWindowAllAnnotations
+
+    companion object {
+        private var requestCode = 0
+
+        @BeforeClass
+        @JvmStatic
+        fun setUpForClass() {
+            // TODO Reflection on Kotlin top-level properties?
+            requestCode = 7
```
Comment (shiraji) [inline test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: I really don't like this code but I'm not sure how to get private top level properties using kotlin-reflection...

**Comment by hotchemi** [comment]
👀

File: test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt (hunk 1 above)
Comment (mannodermaus) [reply test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: Yeah, I had the same problem once, no idea.

**Comment by mannodermaus** [comment]
👀

File: test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt (hunk 1 above)
Comment (shiraji) [reply test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: I asked the question to reflect channel in Kotlinlang

https://kotlinlang.slack.com/archives/C0AVAB92L/p1512281918000028

let's see

Comment (mannodermaus) [reply test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: TIL of the `#reflect` channel on kotlinlang's Slack. Nice!

Comment (shiraji) [reply test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: well, I think at this moment there is no way to solve this problem...

**Comment by hotchemi** [comment]
Ah sorry I'll check again🙇

File: test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt (hunk 1 above)
Comment (mannodermaus) [reply test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: Did you open an issue on YouTrack, by the way? If so, could you share the URL?

**Comment by shiraji** [comment]
I'm not sure why CI failed, thou

File: test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt (hunk 1 above)
Comment (shiraji) [reply test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: Not yet, but I just notice I don't even know how to access a private top property. I read some article that says there is no way to access private top property. I couldn't find it at this moment.

...well, let me open the issue.

Comment (shiraji) [reply test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: https://youtrack.jetbrains.com/issue/KT-21707

Comment (shiraji) [reply test/src/test/java/permissions/dispatcher/test/ActivityWithSystemAlertWindowAllAnnotationsPermissionsDispatcherTest.kt]: Sadly, 

```
there is no way to access private top property
```

this statement is correct... 😢 

https://youtrack.jetbrains.com/issue/KT-21707#comment=27-2595536

**Comment by hotchemi** [comment]
Umm...I tried to rebuild and it passed...may be we need to reset something after test passed? 🤔
Anyway, let me merge and investigate the problem separately~! Thanks!

**Comment by hotchemi** [comment]
Ah ok may be test being failed when multiple processes execute test simultaneously... 🤔

**Pull request merged**